### PR TITLE
[Final][CHIM-2] Additional ease types to allow for back easing

### PR DESCRIPTION
**Pull request by JayAustria**

### CODE REVIEW

**Comment by jeiting** [comment]
Looks great. Only thing, could you update the ease code to be in moai conventions? I'd really like to throw this back at Moai.

File: src/uslscore/USInterpolate.cpp [hunk 1]
```diff
@@ -76,8 +76,46 @@ float USInterpolate::Curve ( u32 mode, float t ) {
...
+			return t * t * ( ( overshoot + 1 ) * t - overshoot );
+		}
+		case kBackSmooth:
+		{
+			const float overshoot = 1.70158f * 1.525f;
+			
+			t = t * 2;
+			
+			if ( t < 1 ) {
+				return ( t * t * ( ( overshoot + 1 ) * t - overshoot )) / 2;
```
Comment (jeiting) [inline src/uslscore/USInterpolate.cpp:98]: overshoot )) 
should be 
overshoot ) )

File: src/uslscore/USInterpolate.cpp [hunk 2]
```diff
@@ -76,8 +76,46 @@ float USInterpolate::Curve ( u32 mode, float t ) {
...
+			}
+		}
+			
+		case kSineEaseIn:
+			
+			return sinf ( t * (float)M_PI_2 );
+			
+		case kSineEaseOut:
+			
+			return -1 * cosf ( t * (float)M_PI_2 ) + 1;
```
Comment (jeiting) [inline src/uslscore/USInterpolate.cpp]: ( float ) M_PI_2 i think,

unless there is a counter example.

Comment (JayAustria) [inline src/uslscore/USInterpolate.cpp]: Examples of both found in MOAI code but ( float ) has more hits. Will change.

File: src/uslscore/USInterpolate.cpp (hunk 1 above)
Comment (JayAustria) [inline src/uslscore/USInterpolate.cpp:98]: No example of two not-touching ending parentheses in this file but there are a few touching end parentheses. How silly.

**Comment by jeiting** [comment]
Tums Uppp

**Pull request merged**